src/plugins/data/common/datasets/types.ts

```diff
@@ -28,6 +28,8 @@ export interface DataSourceMeta {
   name?: string;
   /** Optional session ID for faster responses when utilizing async query sources */
   sessionId?: string;
+  /** Optional supportsTimeFilter determines if a time filter is needed */
+  supportsTimeFilter?: boolean;
```

Comment:
can this be controlled based on whether `timeField` is present in the dataset rather than additional flag?

Reply:
Today even if `timeField` is present we don't have support for S3 datasources to allow time filter injection in the query. This is surely where we want to go towards in future though.

Reply:
Some older conversations around this: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337#discussion_r1779017573, https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337#discussion_r1786959825